### PR TITLE
fix getRandomPort

### LINKED ISSUE #1850: tri协议暴露服务，随机端口不可用
image
port <= 0的时候无法成立，应该转Int再比较？

### PR DESCRIPTION
**Pull request by wujincheng2333**
**What this PR does**: 

**Which issue(s) this PR fixes**: 
Fixes #1850

**You should pay attention to items below to ensure your pr passes our ci test**
We do not merge pr with ci tests failed

- [x] All ut passed (run 'go test ./...' in project root)
- [ ] After go-fmt ed , run 'go fmt project' using goland.
- [ ] Golangci-lint passed, run 'sudo golangci-lint run' in project root.
- [ ] After import formatted, (using [imports-formatter](https://github.com/dubbogo/tools#5-how-to-get-imports-formatter) to run 'imports-formatter .' in project root, to format your import blocks, mentioned in [CONTRIBUTING.md](https://github.com/apache/dubbo-go/blob/master/CONTRIBUTING.md) above) 
- [ ] Your new-created file needs to have [apache license](https://raw.githubusercontent.com/dubbogo/resources/master/tools/license/license.txt) at the top, like other existed file does.
- [ ] All integration test passed. You can run integration test locally (with docker env). Clone our [dubbo-go-samples](https://github.com/apache/dubbo-go-samples) project and replace the go.mod to your dubbo-go, and run 'sudo sh start_integration_test.sh' at root of samples project root. (M1 Slice is not Support)

### CODE REVIEW

**Comment by LaurenceLiZhixin** [comment]
@wujincheng2333 修改base分支到3.0

**Comment by wujincheng2333** [comment]
> @wujincheng2333 修改base分支到3.0

已修改

File: config/service_config.go [hunk 1]
```diff
@@ -247,7 +253,17 @@ func (s *ServiceConfig) Export() error {
 		if len(proto.Port) == 0 {
 			port = nextPort.Value.(string)
 			nextPort = nextPort.Next()
+		}else{
```
Comment (AlexStocks) [inline config/service_config.go:256]: 这代码显然没有 gofmt 处理过

Comment (wujincheng2333) [reply config/service_config.go:256]: > 这代码显然没有 gofmt 处理过

是的，不好意思，我以为你们会修改代码风格之类的

**Pull request closed without merging**